.github/workflows/release.yml

```diff
@@ -0,0 +1,123 @@
+name: Release Actions
+
+env:
+  BOT_USER: heroku-linguist[bot]
+  BOT_EMAIL: 136119646+heroku-linguist[bot]@users.noreply.github.com
+  CARGO_TERM_COLOR: always
+
+on:
+  workflow_dispatch:
+    inputs:
+      bump:
+        description: "Bump"
+        required: true
+        default: 'patch'
+        type: choice
+        options:
+          - major
+          - minor
+          - patch
+
+jobs:
+  prepare:
+    name: Release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Get token for GH application (Linguist)
+        uses: heroku/use-app-token-action@main
+        id: generate-token
+        with:
+          app_id: ${{ vars.LINGUIST_GH_APP_ID }}
+          private_key: ${{ secrets.LINGUIST_GH_PRIVATE_KEY }}
+
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          token: ${{ steps.generate-token.outputs.app_token }}
+
+      - name: Configure git
+        run: |
+          git config --global user.name ${{ env.BOT_USER }}
+          git config --global user.email ${{ env.BOT_EMAIL }}
+
+      - name: Update Rust toolchain
+        run: rustup update
+
+      - name: Rust cache
+        uses: Swatinem/rust-cache@v2.5.1
+
+      - name: Install cargo-bump
+        run: cargo install cargo-bump
+
+      - name: Get previous release version
+        id: previous-version
+        env:
+          GH_TOKEN: ${{ steps.generate-token.outputs.app_token }}
+        run: echo "value=$((gh release view --json tagName --jq '.tagName' 2> /dev/null || echo "0.0.0") | sed 's/^v//')" >> $GITHUB_OUTPUT
+
+      - name: Bump version
+        run: |
+          cargo bump ${{ steps.previous-version.outputs.value }}
+          cargo bump ${{ inputs.bump }}
```

Comment:
The `|| echo "0.0.0"` fallback I think adds a failure mode that would result in the older tags being overwritten if the `gh release` command were to fail for say connectivity reasons?

eg imagine:
1. `gh release` fails, even though there is an existing release (eg `1.2.0`), due to network issues or GitHub having a bad day
2. Instead of the workflow aborting/failing, the `|| echo "0.0.0"` fallback means `value` is set to `0.0.0`
3. As such the first bump command ends up being `cargo bump 0.0.0`, meaning the bump following it (eg `cargo bump minor`) would then turn the version into `0.1.0` - even though `0.1.0` is actually older than the current `1.2.0` release.

To avoid this, either:
1. The implementation needs to tell the difference between "there are no existing releases" vs "the gh release command failed for a different reason". (Which is probably more hassle than it's worth to implement)
2. Or, we just remove the `|| echo "0.0.0"` and accept the fact that the implementation here assumes there is an existing release.

Given that we know an existing release exists for this repo, I think (2) might be the easier option?

Also, I think the `2> /dev/null` might (a) be unnecessary (given that pipes and redirects only redirect stdout by default), and (b) might be best removed so if any error occurs it appears in the GitHub Actions log.

Reply:
yes, i was trying to accommodate the "no previous release" edge case but we know there is one already so i'll drop this extra logic.